pallets/runtime/src/general_tm.rs

```diff
@@ -299,6 +454,119 @@ impl<T: Trait> Module<T> {
 
         Ok(())
     }
+
+    fn unsafe_modify_default_trusted_claim_issuer(ticker: Ticker, trusted_issuer: IdentityId) {
+        TrustedClaimIssuer::mutate(ticker, |identity_list| {
+            if identity_list.contains(&trusted_issuer) {
```

Comment:
We don't need to do this check. We can instead directly pass `is_add_call` to this function and do the if-else on the basis of that.

In the current implementation, if you call `remove_trusted_issuer` and the issuer is not present in the list then instead of being a no-op, it will add that issuer to the list. We don't want that.

Reply:
I don't think so, We have the check for that.
```
ensure!(
     !Self::trusted_claim_issuer(&ticker).contains(&trusted_issuer) && is_add_call,
     Error::<T>::IncorrectOperationOnTrustedIssuer
);
```

Reply:
There was a bug in the above code. `&& should have been ==`. It's now fixed :).
That being said, this optimization can still be done where you pass `is_add_call` to `unsafe_modify_default_trusted_claim_issuer` function and then you don't have to do `if identity_list.contains` which iterates over the whole array.